### PR TITLE
Reloads cart on window focus - fixes #54

### LINKED ISSUE #54: Multiple tab usage makes for erratic cart behaviour
I and presumably many other people often shop with multiple products open in multiple tabs. If you do this however with ngCart the data doesn't stay synchronised between.

Example:

[Tab One] 3x Item A
[Tab Two] 3x item A

__While in tab two we decide to add one of item B__

[Tab One] 3x Item A
[Tab Two] 3x item A, 1x item B

__We go to tab one where one of item C is open and add that to cart__

[Tab One] 3x Item A, 1x item C
[Tab Two] 3x item A, 1x item B

Now we go to checkout the cart contains...

[Checkout - Either tab] 3x Item A, 1x item C

Obviously this can get a lot more complexed and confusing. I've found a solution, I'll do a pull request to accompany this. We basic restore when the tab is focused then run a digest to make sure the appropriate scopes update.

### PR DESCRIPTION
**Pull request by atomworks**
This prevents cart content becoming desynchronised when shop used over multiple tabs. This is achieved by simply re-triggering the load from localStorage mechanism when you refocus a tab.

### CODE REVIEW

File: dist/ngCart.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ angular.module('ngCart', ['ngCart.directives'])
         };
     })
 
-    .run(['$rootScope', 'ngCart','ngCartItem', 'store', function ($rootScope, ngCart, ngCartItem, store) {
+    .run(['$rootScope', '$window', 'ngCart', 'ngCartItem', 'store', function ($rootScope, $window, ngCart, ngCartItem, store) {
```
Comment (snapjay) [inline dist/ngCart.js]: This is already in from your last PR

File: dist/ngCart.js [hunk 2]
```diff
@@ -182,6 +195,7 @@ angular.module('ngCart', ['ngCart.directives'])
 
 
         this.$restore = function(storedCart){
+            console.log('restore', storedCart);
```
Comment (snapjay) [inline dist/ngCart.js]: No console.log in production

File: dist/ngCart.js [hunk 3]
```diff
@@ -25,6 +25,19 @@ angular.module('ngCart', ['ngCart.directives'])
             ngCart.init();
         }
 
+        angular.element($window).bind('focus', function(){
+
+            // if()
+            // console.log("FOCUS", ngCart.isEmpty());
+            console.log('fcus', $rootScope);
```
Comment (snapjay) [inline dist/ngCart.js]: Please remove commented code and console.logs

**Comment by atomworks** [comment]
As this is pretty old, I'll close it up. It's been a while so although I can't remember why, I'm sorry I couldn't follow through on this at the time.

**Pull request closed without merging**